test-bench-utils/lib/content/xxe.js

```diff
@@ -1,9 +0,0 @@
-module.exports.attackXml = `
-<!DOCTYPE read-fs [<!ELEMENT read-fs ANY >
-<!ENTITY passwd SYSTEM "file:///etc/passwd" >]>
```

Comment:
this should be what is dynamic right? the rest should be static and not from a request. customers wouldn't be passing an entire xmlstring. it'd most likely just be the entity

Reply:
I really doubt they'd just be able to pass the entity in most cases, entities have to be declared in the DOCTYPE of an XML doc (afaik), which I doubt where user controlled strings would end up.  most examples of this vuln are usually just uploading an XML file via a form (which is what juiceshop does) - but that doesn't work w Assess b/c we don't wrap the resulting payload as a source (I raised that in SU yesterday).

I can do this, but personally don't feel the need to add the complexity.  The rule is user input + the noent flag set, which this does.

Reply:
got it. it's fine to keep as is

Reply:
> but that doesn't work w Assess b/c we don't wrap the resulting payload as a source (I raised that in SU yesterday).
can you log a ticket for this.  I think we need to emit source events for _every_ file that gets parsed by our multipart form parsers(multer, formidable, etc)